src/back/space/src/api/space.cpp

```diff
@@ -0,0 +1,161 @@
+#include "../../../shared/errors.hpp"
+#include "../model/space_serialize.hpp"
+
+namespace svetit::space::handlers {
+
+Space::Space(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Space::HandleRequestJsonThrow(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body,
+	server::request::RequestContext&) const
+{
+	formats::json::ValueBuilder res;
+
+	try {
+		switch (req.GetMethod()) {
+			case server::http::HttpMethod::kGet:
+				return Get(req, res);
+			case server::http::HttpMethod::kPost:
+				return Post(req, body, res);
+			case server::http::HttpMethod::kDelete:
+				return Delete(req, res);
+			case server::http::HttpMethod::kHead:
+				return Head(req, res);
+			default:
+				throw std::runtime_error("Unsupported");
+				break;
+		}
+	} catch(const errors::Unauthorized& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kUnauthorized);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::BadRequest& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::NotFound& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kNotFound);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::Conflict& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kConflict);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const std::exception& e) {
+		LOG_WARNING() << "Fail to process space handle with method: "
+			<< req.GetMethodStr() << " err: " << e.what();
+		res["err"] = "Fail to process space";
+		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
+	}
+
+	return res.ExtractValue();
+}
+
+formats::json::Value Space::Get(
+	const server::http::HttpRequest& req,
+	formats::json::ValueBuilder& res) const
+{
+	const auto& userId = req.GetHeader(headers::kUserId);
+	if (userId.empty())
+		throw errors::Unauthorized{};
+
+	if (req.HasArg("id"))
+	{
+		const auto id = req.GetArg("id");
+		res = _s.GetById(id, userId);
+	}
+	else if (req.HasArg("key"))
+	{
+		const auto& key = req.GetArg("key");
+		if (key != userId && !_s.CheckKeyByRegex(key))
+			throw errors::BadRequest{"Key must be valid"};
+		res = _s.GetByKey(key, userId);
+	}
+	else if (req.HasArg("link"))
+	{
+		const auto linkId = req.GetArg("link");
+		res = _s.GetByLink(linkId);
+	}
+	else
+		throw errors::BadRequest{"No arguments"};
+
+	return res.ExtractValue();
+}
+
+formats::json::Value Space::Delete(
+	const server::http::HttpRequest& req,
+	formats::json::ValueBuilder& res) const
+{
+	const auto& userId = req.GetHeader(headers::kUserId);
+	if (userId.empty())
+		throw errors::Unauthorized{};
+
+	const auto& id = req.GetArg("id");
+	if (id.empty())
+		throw errors::BadRequest{"Param id must be set"};
```

Comment:
Вынеси пожалуйста куда то в shared и используй везде где параметр UUID , в том числе где передаётся параметр link
```suggestion
	const auto id = parseUUID(req, "id");
```

Где-то в shared в .cpp
```c++
boost::uuids::uuid parseUUID(const server::http::HttpRequest& req, const std::string& key)
{
	try {
		return utils::BoostUuidFromString(req.GetArg(key));
	} catch (const std::exception& e) {
		const auto msg = fmt::format("Param {} parse err: {}", key, e.what());
		throw errors::BadRequest(msg);
	}
	return {};
}
```

Reply:
Готово (везде).